src/context/DynamicFieldsContext.tsx

```diff
@@ -32,21 +36,24 @@ export const DynamicFieldsProvider = ({
     ...defaultDynamicFieldsData,
   });
 
+  const fetchDynamicFields = async () => {
+    const res = await DynamicFieldAPI.getFields();
+    setDynamicFields({
+      childDynamicFields: res.child_fields,
+      guestDynamicFields: res.guest_fields,
+      parentDynamicFields: res.parent_fields,
+      sessionDynamicFields: res.session_fields,
+    });
+  };
+
   useEffect(() => {
-    async function fetchDynamicFields() {
-      const res = await DynamicFieldAPI.getFields();
-      setDynamicFields({
-        childDynamicFields: res.child_fields,
-        guestDynamicFields: res.guest_fields,
-        parentDynamicFields: res.parent_fields,
-        sessionDynamicFields: res.session_fields,
-      });
-    }
     fetchDynamicFields();
   }, []);
 
   return (
-    <DynamicFieldsContext.Provider value={dynamicFields}>
+    <DynamicFieldsContext.Provider
+      value={{ dynamicFields, fetchDynamicFields }}
```

Comment:
`fetchDynamicFields` is provided so that child components can ensure the context is updated when fields are changed